src/common/Guardian.sol

```diff
@@ -50,8 +50,8 @@ contract Guardian is IGuardian {
 
     // --- Admin actions ---
     /// @inheritdoc IGuardian
-    function createPool(address admin, AssetId currency) external onlySafe returns (PoolId poolId) {
-        return hub.createPool(admin, currency);
+    function createPool(uint48 poolId_, address admin, AssetId currency) external onlySafe returns (PoolId poolId) {
```

Comment:
Curious, why finally going with giving the poolId?

Reply:
@mustermeiszer 's proposal to make pool creation more deterministic and easier to batch actions

Reply:
Don't we found and issue with that @mustermeiszer? Some attacker could front run your poolId and prevent you to create any pool?

Reply:
Yes, but that can also happen with the onchain one. Need to resubmit in any case. This makes it easier for the UI though

Reply:
But if we use `+1`, nobody can frontrun the id.

Reply:
The UI can:
- createPool
- read the Event and show the successful creation of the pool with the new PoolId
- batch any number of actions